### DISCUSSION TITLE
Adapting bidi to pdfTeX and LuaTeX

### DISCUSSION THREAD

**Discussion by Udi-Fogiel** [kvafa/bidi#30] [General]
I've started to adapt bidi to pdfTeX. The main work would be anything related to the backend (hyperlinks, colors etc.) and an appropriate way to handle counters. 

Most of the code of bidi can already be used for pdfTeX, so I thought it would be redundant to create a new package. Making bidi pdfTeX aware would be much easier. The current problem I'm facing is that, to my knowledge, this repository is not the most up to date. I saw a few mentions that you (@kvafa ) have a local repository on your machine. 

I think for such a project, some sort of collaboration will be the best way to go, if you will agree of curse.

**Comment by Udi-Fogiel** [comment]
@kvafa I was hoping to hear from you, but I need that for an upcoming project. I'm going to release a package that will use most of the inner files of `bidi`  soon, crediting your work. Hopefully the package will be integrated into `bidi` eventually.

**Comment by ghost** [comment]
Ok, I finally found it. I sincerely apologize from the bottom of my heart for replying so late. Yes, this is a wonderful idea. The port to PDFTeX should not require much effort since in theory both PDFTeX and XeTeX use e-TeX which itself uses a slightly modified version of TeX--XeT algorithm for bidirectional typesetting.

The port to LuaTeX should also be straightforward. The major part is implementing a unicode bidirectional algorithm. As I mentioned elsewhere, I have experimented with LuaTeX and I have a clear idea of what needs to be done.

If you are still interested and have not lost interest, then please let me know.
